CONTRIBUTING.md

```diff
@@ -125,3 +137,90 @@ will not be merged.
+
+#### Triage steps
+
+1. If `pyproject.toml` is missing or `-vvv` debug logs (with stack trace) is not provide and required, request that the issue author provide it.
+1. Attempt to reproduce the issue with reported version or request further clarification from the issue author.
+1. Ensure the issue is not already resolved. You can attempt to reproduce using latest preview release and/or poetry from the main branch.
+1. If issue cannot be reproduced,
+   1. clarify with issue author,
+   1. close issue or notify `@python-potry/triage`.
+1. If issue can be reproduced,
+   1. comment on the issue confirming so
+   1. notify `@python-potry/triage`.
+   1. if possible, identify root cause of the issue.
+   1. if interested, attempt to fix via a pull request.
+
+#### Multiple versions
+
+Often times you would want to attempt to reproduce issues with multiple versions of `poetry` at the same time. For these use cases, the [pipx project](https://pipxproject.github.io/pipx/) is useful.
+
+You can set your environment up like so.
+
+```sh
+pipx install --suffix @1.0.10 'poetry==1.0.10'
+pipx install --suffix @1.1.0rc1 'poetry==1.1.0rc1'
+pipx install --suffix @master 'poetry @ git+https://github.com/python-poetry/poetry'
+```
+
+> **Hint:** Do not forget to update your `poetry@master` installation in sync with upstream.
+
+For `@local` it is recommended that you do something similar to the following as editable installs are not supported for PEP 517 projects.
+
+```sh
+# note this will not work for Windows, and we assume you have already run `poetry install`
+cd /path/to/python-poetry/poetry
+ln -sf $(poetry run which poetry) ~/.local/bin/poetry@local
+```
+
+> **Hint:** This mechanism can also be used to test pull requests.
+
+### Git Workflow
+
+All development work is performed against Poetry's main branch (`master`). All changes are expected to be submitted and accepted to this
+branch.
+
+#### Release branch
+
+When a release is ready, the following are required before a release is tagged.
+
+1. A release branch with prefix `release-`, eg: `release-1.1.0rc1`.
+1. A pull request from release branch to main branch (`master`) if minor/major release. Otherwise, to bug fix branch (eg: `1.0`).
+   1. The pull request description MUST include change log corresponding to the release (eg: [#2971](https://github.com/python-poetry/poetry/pull/2971)).
+   1. The pull request must contain a commit that updates [CHANGELOG.md](CHANGELOG.md) and bumps project version (eg: [#2971](https://github.com/python-poetry/poetry/pull/2971/commits/824e7b79defca435cf1d765bb633030b71b9a780)).
+   1. The pull request must have a `Release` label specified.
+
+Once the branch pull-request is ready and approved, a member of `@python-poetry/core` will,
+
+1. Tag the branch with the version identifier (eg: `1.1.0rc1`).
+2. Merge the pull request once the release is created and assets uploaded by CI.
```

Comment:
```suggestion
1. A release branch with the prefix `release-`, eg: `release-1.1.0rc1`.
1. A pull request from the release branch to the main branch (`master`) if it's a minor or major release. Otherwise, to the bug fix branch (eg: `1.0`).
   1. The pull request description MUST include the change log corresponding to the release (eg: [#2971](https://github.com/python-poetry/poetry/pull/2971)).
   1. The pull request must contain a commit that updates [CHANGELOG.md](CHANGELOG.md) and bumps the project version (eg: [#2971](https://github.com/python-poetry/poetry/pull/2971/commits/824e7b79defca435cf1d765bb633030b71b9a780)).
   1. The pull request must have the `Release` label specified.

Once the branch pull-request is ready and approved, a member of `@python-poetry/core` will,

1. Tag the branch with the version identifier (eg: `1.1.0rc1`).
2. Merge the pull request once the release is created and assets are uploaded by the CI.
```